x/shared/types/params.go

```diff
@@ -206,3 +247,34 @@ func validateGracePeriodOffsetBlocksIsLessThanNumBlocksPerSession(params *Params
 	}
 	return nil
 }
+
+// validateSupplierUnbondingPeriodIsGreaterThanCumulatedProofWindowCloseBlocks
+// validates that the SupplierUnbondingPeriod blocks is greater than the cumulated
+// proof window close blocks.
+// It ensures that a supplier cannot unbond before the pending claims are settled.
+func validateSupplierUnbondingPeriodIsGreaterThanCumulatedProofWindowCloseBlocks(params *Params) error {
+	cumulatedProofWindowCloseBlocks := GetCumulatedProofWindowCloseBlocks(params)
+	supplierUnbondingPeriodBlocks := params.SupplierUnbondingPeriod * params.NumBlocksPerSession
+
+	if supplierUnbondingPeriodBlocks < cumulatedProofWindowCloseBlocks {
+		return ErrSharedParamInvalid.Wrapf(
+			"SupplierUnbondingPeriod (%v session) (%v blocks) must be greater than the cumulated ProofWindowCloseOffsetBlocks (%v)",
+			params.SupplierUnbondingPeriod,
+			supplierUnbondingPeriodBlocks,
+			cumulatedProofWindowCloseBlocks,
+		)
+	}
+
+	return nil
+}
+
+// GetCumulatedProofWindowCloseBlocks returns the total number of blocks required
+// to pass the proof window close of a session.
+// Using shared.GetProofWindowCloseOffsetHeight is not possible because of the
```

Comment:
Is there a way to just keep this function in either one of `shared` or `session` and (for example) have session import this? We'd delete the other function.

Reply:
`shared` package depends on `types` and `session` depends on `shared`.

Since `GetProofWindowCloseOffsetHeight` uses all the `GetXXXWindowXXXOffsetHeight` functions, we'll need to move all of them the `types` package.

This function is equivalent to calling `shared.GetProofWindowCloseOffsetHeight(params, 0)`, So I'm really tempted to move `x/shared/session.go` to `x/shared/types/session.go` :thinking: